File: FestivalC#/Festival/Services/Service/ModelServices/ServiceUser.cs
using Festival.Model;
using Festival.Repository.Interface;
using Festival.Validation;
using System.Collections.Generic;

namespace Festival.Service
{
    public class ServiceUser : IServiceUser
    {
        //repo
        private IUserRepository repository;

        //validator
        private ValidatorUser validator;

        //constructor
        public ServiceUser(IUserRepository repository, ValidatorUser validator)
        {
            this.repository = repository;
            this.validator = validator;
        }

        public void save(User user)
        {
            repository.save(user);
        }

        public void delete(string username)
        {
            repository.delete(username);
        }

        public void update(string username, User user)
        {
            repository.update(username, user);
        }

        public User getById(string username)
        {
            return repository.getById(username);
        }

        public List<User> getAll()
        {
            return repository.getAll();
        }

        public User getByNamePass(string username, string password)
        {
            return repository.getUserFor(username, password);
        }
    }
}

File: FestivalJava/Services/src/main/java/festival/ModelServices/ServiceUser.java
package festival.ModelServices;

import festival.Domain.User;
import festival.Interfaces.IUserRepository;
import festival.ModelServices.Interfaces.IServiceUser;
import festival.Validation.Exceptions.ValidatorException;
import festival.Validation.ValidatorUser;

import java.util.List;

/**
 * Created by Sebi on 26-Mar-17.
 */
public class ServiceUser implements IServiceUser{
    //repository
    private IUserRepository repository;

    //validator
    private ValidatorUser validator;

    //constructor
    public ServiceUser(IUserRepository repository, ValidatorUser validator){
        this.repository = repository;
        this.validator = validator;
    }

    public void save(User user) throws ValidatorException {
        validator.validate(user);
        repository.save(user);
    }

    public void delete(String username)
    {
        repository.delete(username);
    }

    public void update(String username, User user) throws ValidatorException {
        validator.validate(user);
        repository.update(username, user);
    }

    public User getById(String username)
    {
        return repository.getById(username);
    }

    public List<User> getAll()
    {
        return repository.getAll();
    }

    public User getByNamePass(String username, String password)
    {
        return repository.getUserForLogin(username, password);
    }
}
